src/Libraries/Microsoft.Extensions.AI.Abstractions/Functions/AIFunctionArguments.cs

```diff
@@ -20,36 +21,42 @@ namespace Microsoft.Extensions.AI;
 /// an <see cref="AIFunction"/> if it needs to resolve any services from a dependency injection
 /// container.
 /// </remarks>
-public sealed class AIFunctionArguments : IDictionary<string, object?>, IReadOnlyDictionary<string, object?>
+public class AIFunctionArguments : IDictionary<string, object?>, IReadOnlyDictionary<string, object?>
 {
     /// <summary>The nominal arguments.</summary>
     private readonly Dictionary<string, object?> _arguments;
 
     /// <summary>Initializes a new instance of the <see cref="AIFunctionArguments"/> class.</summary>
-    public AIFunctionArguments()
+    /// <param name="comparer">The optional <see cref="IEqualityComparer{T}"/> to use for key comparisons.</param>
+    public AIFunctionArguments(IEqualityComparer<string>? comparer = null)
```

Comment:
This would be a binary breaking change. I recommend we add this by way of a new overload.

Reply:
What's driving the need to override the comparison? Is this based on wanting to use OrdinalIgnoreCase for SK?